include/bluetooth/hci_raw.h

```diff
@@ -28,7 +28,12 @@ extern "C" {
 #endif /* CONFIG_BT_CTLR */
 
 /** Data size needed for ACL buffers */
+#if defined(CONFIG_BT_BUF_ACL_DATA_LEN)
+#define BT_BUF_ACL_SIZE MAX(BT_L2CAP_BUF_SIZE(BT_L2CAP_MTU), \
+	(CONFIG_BT_BUF_ACL_DATA_LEN + BT_HCI_ACL_HDR_SIZE + BT_BUF_RESERVE))
+#else
 #define BT_BUF_ACL_SIZE BT_L2CAP_BUF_SIZE(BT_L2CAP_MTU)
```

Comment:
Just curious, when would this apply? You include hci_raw.h but don't enable BT_HCI_RAW. Could we reduce the amount of possibilities for the BT_BUF_ACL_SIZE and end up with less surprises?

Reply:
Maybe we could eliminate the define above? The ACL shouldn't have to depend on the L2CAP MTU, because that should be fragmented across the LL PDUs. The ACL should be aligned with the LL PDU size like it is when CONFIG_BT_CTLR_TX_BUFFER_SIZE is defined.

It would be good if we could make the buffer size configurations simpler because we end up with many of these situations where depending on the board selected we get different buffer sizes.

Reply:
> @jostn Could you have a look at my questions, so that we can proceed?



> @jostn Could you have a look at my questions, so that we can proceed?

I've reworked the PR so that we do not depend on l2cap any longer.